resource/event_log.go

```diff
@@ -0,0 +1,840 @@
+	eventLogInfo.UefiEventAddr = 0
+	eventLogInfo.UefiEventSize = 0
+	eventLogInfo.TxtEnabled = false
+	return eventLogInfo
+}
+
+//GetEventLogInfo - Getter function for eventLogInfo variable
+func GetEventLogInfo() *EventLogInfo {
+	log.Trace("resource/event_log:GetEventLogInfo() Entering")
+	defer log.Trace("resource/event_log:GetEventLogInfo() Leaving")
+	if eventLogInfo == nil {
+		eventLogInfo = InitializeEventLogInfo()
+	}
+	return eventLogInfo
+}
+
+//FetchUefiEventInfo - Function to get Uefi Events Info
+func (eventLogInfo *EventLogInfo) FetchUefiEventInfo() error {
+
+	log.Trace("resource/event_log:FetchUefiEventInfo() Entering")
+	defer log.Trace("resource/event_log:FetchUefiEventInfo() Leaving")
+
+	uefiBaseAddress := make([]byte, Uint64Size)
+	uefiSize := make([]byte, Uint32Size)
+	file, err := os.Open(Tpm2FilePath)
+	if err != nil {
+		return errors.Wrapf(err, "resource/event_log:FetchUefiEventInfo() There was an error opening %s", Tpm2FilePath)
+	}
+	defer file.Close()
+
+	fileInfo, err := file.Stat()
+	if err != nil {
+		return errors.Wrapf(err, "resource/event_log:FetchUefiEventInfo() There was an error getting FileInfo of %s", Tpm2FilePath)
+	}
+
+	// As TPM2 file contains uefi_event_log size from offset 64-67 and uefi_event_log address from offset 68-75, so below check is for uefi_event_log Info in TPM2 file
+	if fileInfo.Size() < 76 {
+		secLog.Errorf("resource/event_log:FetchUefiEventInfo() UefiEventInfo missing in %s", Tpm2FilePath)
+		return errors.Errorf("UefiEventInfo missing in %s", Tpm2FilePath)
+	}
+
+	_, err = file.Seek(UefiBaseOffset, io.SeekStart)
+	if err != nil {
+		return errors.Wrapf(err, "resource/event_log:FetchUefiEventInfo() There was an error traversing %s", Tpm2FilePath)
+	}
+	_, err = io.ReadFull(file, uefiBaseAddress)
+	if err != nil {
+		return errors.Wrapf(err, "resource/event_log:FetchUefiEventInfo() There was an error reading %s", Tpm2FilePath)
+	}
+	_, err = file.Seek(UefiSizeOffset, io.SeekStart)
+	if err != nil {
+		return errors.Wrapf(err, "resource/event_log:FetchUefiEventInfo() There was an error traversing %s", Tpm2FilePath)
+	}
+	_, err = io.ReadFull(file, uefiSize)
+	if err != nil {
+		return errors.Wrapf(err, "resource/event_log:FetchUefiEventInfo() There was an error reading %s", Tpm2FilePath)
+	}
+
+	eventLogInfo.UefiEventSize = binary.LittleEndian.Uint32(uefiSize)
+	eventLogInfo.UefiEventAddr = binary.LittleEndian.Uint64(uefiBaseAddress)
+	return nil
+}
+
+//FetchTxtHeapInfo - Function to fetch txt events info
+func (eventLogInfo *EventLogInfo) FetchTxtHeapInfo() error {
+	log.Trace("resource/event_log:FetchTxtHeapInfo() Entering")
+	defer log.Trace("resource/event_log:FetchTxtHeapInfo() Leaving")
+
+	txtHeapBaseAddr := make([]byte, Uint64Size)
+	txtHeapSize := make([]byte, Uint64Size)
+	file, err := os.Open(EventLogFilePath)
+	if err != nil {
+		return errors.Wrapf(err, "resource/event_log:FetchTxtHeapInfo() There was an error opening %s", EventLogFilePath)
+	}
+	defer file.Close()
```

Comment:
handle errors in defer

Reply:
resolving comment -- file.Close() is handled in a defer func() as recommended.